build-support/bin/external_tool_versions.py

```diff
@@ -0,0 +1,169 @@
+
+        key = key_element.text
+        if key and regex.match(key):
+            yield f"https://cdn.dl.k8s.io/{key}"
+
+
+def get_k8s_versions(url_template: str, pool: ThreadPool) -> Iterator[str]:
+    response = requests.get("https://cdn.dl.k8s.io/", allow_redirects=True)
+    urls = _parse_k8s_xml(response.text)
+    for v in pool.imap_unordered(fetch_text, urls):
+        yield v.strip().lstrip("v")
+
+
+DOMAIN_TO_VERSIONS_MAPPING: dict[str, Callable[[str, ThreadPool], Iterator[str]]] = {
+    # TODO github.com
+    "dl.k8s.io": get_k8s_versions,
+}
+
+
+def fetch_version(url_template: str, version: str, platform: str) -> VersionHash | None:
+    url = url_template.format(version=version, platform=platform)
+    response = requests.get(url, allow_redirects=True)
+    if response.status_code != 200:
+        logger.error("failed to fetch version: %s\n%s", version, response.text)
+        return None
+
+    size = len(response.content)
+    sha256 = hashlib.sha256(response.content)
+    return VersionHash(
+        version=version,
+        platform=platform,
+        size=size,
+        sha256=sha256.hexdigest(),
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-t",
+        "--tool",
+        help="Python tool location, for example: pants.backend.tools.taplo.subsystem:Taplo",
+        required=True,
+    )
+    parser.add_argument(
+        "--platforms",
+        default="macos_arm64,macos_x86_64,linux_arm64,linux_x86_64",
+        help="Comma separated list of platforms",
+    )
+    parser.add_argument(
+        "-w",
+        "--workers",
+        default=16,
+        help="Thread pool size",
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+        help="Verbose output",
+    )
+
+    args = parser.parse_args()
+
+    level = logging.DEBUG if args.verbose else logging.INFO
+    logging.basicConfig(level=level, format="%(message)s")
+
+    module_string, class_name = args.tool.split(":")
+    module = importlib.import_module(module_string)
+    cls = getattr(module, class_name)
+
+    platforms = args.platforms.split(",")
+    platform_mapping = cls.default_url_platform_mapping
+    mapped_platforms = {platform_mapping.get(p) for p in platforms}
+
+    domain = urlparse(cls.default_url_template).netloc
+    get_versions = DOMAIN_TO_VERSIONS_MAPPING[domain]
+    pool = ThreadPool(processes=args.workers)
+    results = []
+    for version in get_versions(cls.default_url_template, pool):
+        for platform in mapped_platforms:
+            logger.debug("fetching version: %s %s", version, platform)
+            results.append(
+                pool.apply_async(fetch_version, args=(cls.default_url_template, version, platform))
+            )
+
+    backward_platform_mapping = {v: k for k, v in platform_mapping.items()}
+    for result in results:
+        v = result.get(60)
```

Comment:
clarify that this is for the timeout
```suggestion
        v = result.get(timeout=60)
```